org.mixedrealitytoolkit.spatialmanipulation/ObjectManipulator/ManipulationSpecializations/IManipulationSpecialization.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Mixed Reality Toolkit Contributors
+// Licensed under the BSD 3-Clause
+
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace MixedReality.Toolkit.SpatialManipulation
+{
+    /// <summary>
+    /// An interface that allows for specialized manipulation of an <see cref="ObjectManipulator"/>
+    /// under certain circumstances, such as when being selected by an <see cref="XRSocketInteractor"/>,
+    /// that may call for manipulation behavior that is distinct from the default.
+    /// </summary>
+    /// <remarks>
+    /// When active, the manipulation specialization is responsible for complete
+    /// manipulation details, including final position, rotation, and scale of the object
+    /// along with any smoothing and handling of the <see cref="Rigidbody"/> when present.
+    /// A <see cref="BaseManipulationSpecialization"/> is provided for convenience, but implementers
+    /// can use this interface directly as well.
+    /// </remarks>
+    /// <seealso cref="ObjectManipulator"/>
+    public interface IManipulationSpecialization
+    {
+        /// <summary>
+        /// Whether this manipulation specialization is valid for and can process the
+        /// current state of the object with selecting interactors.
+        /// </summary>
+        /// <param name="interactors">
+        /// List of all <see cref="IXRSelectInteractor"/> objects selecting this object.
+        ///</param>
+        /// <param name= "interactable">
+        /// The <see cref="IXRSelectInteractable"/> that is being manipulated.
+        /// <remarks>
+        /// The list of attached specialization components will be queried in order during
+        /// <see cref="ObjectManipulator.OnSelectEntered(SelectEnterEventArgs)"/> and the first to
+        /// respond with <c>true</c> will be used as the active manipulation specialization.
+        /// Otherwise, if no specializations respond, the default object manipulation will be used.
+        /// </remarks>
+        public bool CanProcessSelection(List<IXRSelectInteractor> interactors,
```

Comment:
Can we simplify and reduce the number of parameters? It seems like you only need two parameters `interactors` and `objectmanipulator` and the host transform and rigidBody can be gotten from the object manipulator.